extensions-contrib/first-last-string/src/main/java/io/druid/query/aggregation/first/StringFirstAggregator.java

```diff
@@ -0,0 +1,100 @@
+/*
+ * Licensed to Metamarkets Group Inc. (Metamarkets) under one
+ * or more contributor license agreements. See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership. Metamarkets licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.druid.query.aggregation.first;
+
+import io.druid.collections.SerializablePair;
+import io.druid.query.aggregation.Aggregator;
+import io.druid.segment.BaseLongColumnValueSelector;
+import io.druid.segment.BaseObjectColumnValueSelector;
+
+public class StringFirstAggregator implements Aggregator
+{
+
+  private final BaseObjectColumnValueSelector valueSelector;
+  private final BaseLongColumnValueSelector timeSelector;
+  private final Integer maxStringBytes;
+
+  protected long firstTime;
+  protected String firstValue;
+
+  public StringFirstAggregator(
+      BaseLongColumnValueSelector timeSelector,
+      BaseObjectColumnValueSelector valueSelector,
+      Integer maxStringBytes
+  )
+  {
+    this.valueSelector = valueSelector;
+    this.timeSelector = timeSelector;
+    this.maxStringBytes = maxStringBytes;
+
+    firstTime = Long.MAX_VALUE;
+    firstValue = null;
+  }
+
+  @Override
+  public void aggregate()
+  {
+    long time = timeSelector.getLong();
+    if (time < firstTime) {
+      firstTime = time;
+      Object value = valueSelector.getObject();
```

Comment:
We can add another case if `value` is not String, a SerializablePair and `null` .. we can do a `value.toString`. 

If `value == null`, the aggregator skips it.

Reply:
I mean, `value` might be accidentally another type because of bugs. We need to add a sanity check.

Reply:
Currently, if the value is not String or SerializablePairLongStrin and if it differents of null the aggregator perform a `value.toString()` to transform it to String. What do you think??

Reply:
I think it's better to throw an exception because `value` is always supposed to be a String or a SerializablePair. I'm not sure how it's worth for users to return `value.toString()`. Maybe it's not the result users expect to get from this aggregator.

Reply:
ok! Maybe the exception is a better option.